File: java/src/test/java/problems101_200/RotateArrayTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class RotateArrayTest extends SampledTest {
    @Test
    public void test() {
        for (RotateArray solution : RotateArray.values()) {
            for (RotateArraySample sample : getSamples(RotateArraySample[].class)) {
                int[] nums = sample.input.nums;
                solution.rotate(nums, sample.input.k);
                assertWithMessage(sample.getMessage())
                    .that(nums)
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/RotateArrayTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class RotateArrayTest : SampledTest() {
    @Test
    fun test() =
        RotateArray.entries.forEach { solution ->
            getSamples<Array<RotateArraySample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(input.nums.also { solution.rotate(it, input.k) })
                    .isEqualTo(output)
            }
        }
}
